Tasks/AzureRmWebAppDeployment/AzureRmWebAppDeployment.ps1

```diff
@@ -1,134 +1,142 @@
+	$AdditionalArguments = Get-VstsInput -Name AdditionalArguments 
+	$WebAppUri = Get-VstsInput -Name WebAppUri
+	$SetParametersFile = Get-VstsInput -Name SetParametersFile
+
+	# Initialize Azure.
+
+	Import-Module $PSScriptRoot\ps_modules\VstsAzureHelpers_
+	Initialize-Azure
+
+	# Import the loc strings.
+	Import-VstsLocStrings -LiteralPath $PSScriptRoot/Task.json
+
+
+	Write-Verbose "Starting AzureRM WebApp Deployment Task"
+
+	Write-Verbose "ConnectedServiceName = $ConnectedServiceName"
+	Write-Verbose "WebAppName = $WebAppName"
+	Write-Verbose "DeployToSlotFlag = $DeployToSlotFlag"
+	Write-Verbose "ResourceGroupName = $ResourceGroupName"
+	Write-Verbose "SlotName = $SlotName"
+	Write-Verbose "Package = $Package"
+	Write-Verbose "SetParametersFile = $SetParametersFile"
+	Write-Verbose "RemoveAdditionalFilesFlag = $RemoveAdditionalFilesFlag"
+	Write-Verbose "ExcludeFilesFromAppDataFlag = $ExcludeFilesFromAppDataFlag"
+	Write-Verbose "TakeAppOfflineFlag = $TakeAppOfflineFlag"
+	Write-Verbose "VirtualApplication = $VirtualApplication"
+	Write-Verbose "AdditionalArguments = $AdditionalArguments"
+	Write-Verbose "WebAppUri = $WebAppUri"
+
+	$WebAppUri = $WebAppUri.Trim()
+	$Package = $Package.Trim('"').Trim()
+
+	if( [string]::IsNullOrEmpty($Package) ){
+		Throw (Get-VstsLocString -Key "Invalidwebapppackagepathprovided")
+	}
 
-    [String] [Parameter(Mandatory = $true)]
-    $DeployToSlotFlag,
+	if ($ExcludeFilesFromAppDataFlag -match '[\r\n]' -or [string]::IsNullOrEmpty($ExcludeFilesFromAppDataFlag)) {
+		$ExcludeFilesFromAppDataFlag = $false
+	}
 
-    [String] [Parameter(Mandatory = $false)]
-    $ResourceGroupName,
+	if ($TakeAppOfflineFlag -match '[\r\n]' -or [string]::IsNullOrEmpty($TakeAppOfflineFlag)) {
+		$TakeAppOfflineFlag = $false
+	}
 
-    [String] [Parameter(Mandatory = $false)]
-    $SlotName,
+	if ($RemoveAdditionalFilesFlag -match '[\r\n]' -or [string]::IsNullOrEmpty($RemoveAdditionalFilesFlag)) {
+		$RemoveAdditionalFilesFlag = $false
+	}
 
-    [String] [Parameter(Mandatory = $true)]
-    $Package,
-    
-    [String] [Parameter(Mandatory = $false)]
-    $SetParametersFile,
+	$SetParametersFile = $SetParametersFile.Trim('"').Trim()
 
-    [String] [Parameter(Mandatory = $false)]
-    $RemoveAdditionalFilesFlag,
 
-    [String] [Parameter(Mandatory = $false)]
-    $ExcludeFilesFromAppDataFlag,
+	# Load all dependent files for execution
+	Import-Module ./AzureUtility.ps1 -Force
+	Import-Module ./Utility.ps1 -Force
+	Import-Module ./FindInstalledMSDeploy.ps1
 
-    [String] [Parameter(Mandatory = $false)]
-    $TakeAppOfflineFlag,
+	 # Importing required version of azure cmdlets according to azureps installed on machine
+	 $azureUtility = Get-AzureUtility
 
-    [String] [Parameter(Mandatory = $false)]
-    $VirtualApplication,
+	 Write-Verbose  "Loading $azureUtility"
+	 Import-Module ./$azureUtility -Force
 
-	[String] [Parameter(Mandatory = $false)]
-	[String] $AdditionalArguments,
+	$ErrorActionPreference = 'Stop'
```

Comment:
This is already set by the lib. https://github.com/Microsoft/vsts-task-lib/blob/master/powershell/Docs/ErrorsWarningsAndTaskResult.md